compiler/optimizer/VirtualGuardHeadMerger.cpp

```diff
@@ -47,6 +47,11 @@ static bool isMergeableGuard(TR::Node *node)
    return mergeOnlyHCRGuards ? node->isHCRGuard() : node->isNopableInlineGuard();
    }
 
+static bool isStopTheWorldGuard(TR::Node *node)
+   {
+   return node->isHCRGuard() || node->isOSRGuard();
+   }
+
```

Comment:
While true, this query is incomplete. Profiled guards are also "stop the world" guards as seen in the following comment:
https://github.com/eclipse/omr/blob/28a04889bbb3906fa8dfc9635cfcf4798f95c827/compiler/z/codegen/S390Instruction.cpp#L5792-L5804

Is it possible to add profiled guards to this list? If so that would be great, because we could abstract this API into a common place and reuse it both here and in the codegen location linked above.

Reply:
Profiled guards aren't included in this query as this optimization will ignore them in `isMergeableGuard `. As all guards must first be mergeable to be considered, they weren't a concern. Therefore, using a common API shouldn't modify the behaviour.

Reply:
Profiled guards shouldn't be stop the world, so Z codegen might be mistaken here. A common API to check for stop the world guards should be added though, in a later change.